libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/RouteStepProgress.kt

```diff
@@ -9,15 +9,15 @@ import com.mapbox.geojson.Point
  * The latest route step progress object can be obtained through the [com.mapbox.navigation.base.trip.RouteProgressObserver].
  * Note that the route step progress object's immutable.
  */
-class RouteStepProgress private constructor(
-    private val stepIndex: Int = 0,
-    private val step: LegStep? = null,
-    private val stepPoints: List<Point>? = null,
-    private val distanceRemaining: Float = 0f,
-    private val distanceTraveled: Float = 0f,
-    private val fractionTraveled: Float = 0f,
-    private val durationRemaining: Double = 0.0,
-    private val guidanceViewURL: String? = null,
+data class RouteStepProgress(
```

Comment:
Same as above re: `private`

Reply:
yeah there are reasons, so let's discuss in the other thread: https://github.com/mapbox/mapbox-navigation-android/pull/2992#discussion_r426939789